tests/acceptance/route-transitions-test.ts

```diff
@@ -19,4 +19,31 @@ module('Acceptance | route transitions', function(hooks) {
     assert.dom('#loading-indicator').doesNotExist();
     assert.dom('#loading-content').exists();
   });
+
+  test('router transitions do not affect loading state', async function(assert) {
+
+    let config = this.owner.resolveRegistration('config:environment');
+    config['ember-loading'] = {
+      watchTransitions: false
+    };
```

Comment:
Maybe it's better to lookup the loading service here and directly override the `watchTransitions` property, instead of patching the config? This might help with test leaking issue you mentioned below.